### ISSUE TITLE
Please make the pn.panel chosen pane for a DataFrame awesome

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#5131]
I just engineered a prompt for making chatGPT awesome at converting Streamlit to Panel data apps. But the Panel apps created are not great because the a dataframe is shown using the `pn.widgets.DataFrame` which is not awesome.

Please instead use the `pn.pane.DataFrame` or `pn.pane.Tabulator` or `pn.pane.Perspective` components instead by default. I would actually recommend perspective by default. Because if you use `pn.panel` you are in exploration or quick mode.

Alternatively describe a way to configure the default component used to display a dataframe.

Here is the app

![image](https://github.com/holoviz/panel/assets/42288570/a762719b-4d33-4e42-9b49-c0b1d09d1b83)

```python
import requests
import pandas as pd
import panel as pn

def fetch_emojis():
    resp = requests.get(
        'https://raw.githubusercontent.com/omnidan/node-emoji/master/lib/emoji.json')
    json = resp.json()
    codes, emojis = zip(*json.items())
    return pd.DataFrame({
        'Emojis': emojis,
        'Shortcodes': [f':{code}:' for code in codes],
    })

emojis = fetch_emojis()

intro_text = """
# Panel emoji shortcodes

Below are all the emoji shortcodes supported by Panel.

Shortcodes are a way to enter emojis using pure ASCII. So you can type this `:smile:` to show this
:smile:.

(Keep in mind you can also enter emojis directly as Unicode in your Python strings too — you don't
*have to* use a shortcode)
"""

table = pn.widgets.DataFrame(emojis, width=800, height=400)

app = pn.Column(intro_text, table)

app.show()
```

**Comment by philippjfr** [comment]
Would love to hear other people's thoughts here but personally I think the default DataFrame renderer should be lightweight. So I think the correct approach here is to tweak the CSS rather than switching to some other implementation. 

I find `pn.pane.DataFrame` quite ugly, and I definitely would not want something like Perspective to be the default table renderer. `Tabulator` might be an option but it also requires a bunch of extra resources. Any thoughts @holoviz/panel-maintainers?

**Comment by jbednar** [comment]
Agreed. Perspective or Tabulator are great for when someone needs to dive deeply into a table or to modify it, but I think the default should be something clean, easy to read, and nice looking, and I assume that can be achieved with CSS alone. There are various examples at https://ardata.fr/en/flextable-gallery/ and  https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html for inspiration.

**Comment by maximlt** [comment]
Just all note that in Marc's example he's using the DataFrame **widget**, not pane. `pn.panel` on a dataframe returns the DataFrame **pane**.

> Would love to hear other people's thoughts here but personally I think the default DataFrame renderer should be lightweight.

I think the default DataFrame **pane** is indeed ugly but also too lightweight in features. I'd expect to be at least able to sort the columns :) Or put in a different way, why should I have to go from using a Pane to a Widget when I have data has more than a few rows, because then I would like to be able to filter it/paginate it for easier exploration. I think the difference we make between widgets and panes is that widgets' data can be edited, interactivity isn't just for widgets, e.g. the plot panes often have click/selection events. (Interestingly I just saw that the Perspective pane is editable by default, except the data is not synced back).

> `Tabulator` might be an option but it also requires a bunch of extra resources

Can we quantify how much extra resources it requires?

**Comment by MarcSkovMadsen** [comment]
I would say that if you know your return value or dataframe you can always select a specific Pane. But for exploration and ChatGPT use case the dataframes can be anything including having many columns and rows. There you would like to be able to explore. That is the use case that `pn.panel` should support.

- Small and big dataframes
- Look awesome
- Interactivity like sorting and filtering.
